cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -558,11 +563,22 @@ public void createApplicationVersion(ApplicationId id,
       }
       fields.add(Fields.booleanField(StoreDefinition.AppMetadataStore.LATEST_FIELD, false));
       getApplicationSpecificationTable().upsert(fields);
+      // increment metric : event.deploy.new.count
+      emitMetrics(id.getNamespace(), id.getApplication(), Constants.Metrics.AppMetadataStore.EVENT_DEPLOY_NEW_COUNT);
```

Comment:
Can this be moved to `ApplicationRegistrationStage` ? The only info that is missing seems to be whether it is a deploy or edit. Is it possible to change the `createApplicationVersion` method to return a status object with the necessary details? Existing callers can ignore this and the `ApplicationRegistrationStage` can use this information to emit the metrics. This will also avoid the problem of emitting metrics if there is an exception in `writeApplication `

Reply:
so in that case, do you mean a status indicating if its a new deploy/ edit deploy? So in that case we will have to inject `metricsCollectionService` in `LocalApplicationManager`. @albertshau @tivv do you have any concerns with this ?